test/system/helpers.network.bash

```diff
@@ -332,6 +354,9 @@ function random_free_port_range() {
 function port_is_bound() {
     local port=${1?Usage: port_is_bound PORT [tcp|udp] [ADDRESS]}
 
+    # First make sure no other tests are using it
+    reserve_port $port || return 0
```

Comment:
wait_for_port calls port_is_bound so I don't think it is correct to allocate the port here
It will break wait_for_port for callers such as in the podman-system-servicetests that call random_free_port followed by wait_for_port on the returned port as this would not always return true without actually checking if the port is bound.

Reply:
Yikes, good catch. Thank you. I've pushed a new version, but have marked this as draft until that version gets some heavy testing in #23275.